File: java dsa codes/doublelinklist.java
public class doublelinklist {

    static class node{
        int data;
        node next;
        node prev;
    }

    public static void linklisttraversal(node head){
        node ptr=head;
        while(ptr!=null){
            System.out.println("elements :"+ptr.data);
            ptr=ptr.next;
        }
    }
    
    public static void main(String [] args){
        node head=new node();
        head.data=10;
        head.next=null;
        head.prev=null;
        
        node second=new node();
        second.data=20;
        second.next=null;
        second.prev=null;

        node third=new node();
        third.data=30;
        third.next=null;
        third.prev=null;

        head.next=second;
        second.next=third;
        second.prev=head;
        third.prev=second;

        linklisttraversal(head);
    }

    
}

File: c dsa codes/doublelinklist.c
#include<stdio.h>
#include<stdlib.h>
struct node {
    int data;
    struct node* next;
    struct node* prev;
};

void linklisttraversal(struct node* head){
    struct node* ptr=head;
    while(ptr!=NULL){
        printf("element: %d\n",ptr->data);
        ptr=ptr->next;
    }
}

int main(){
    struct node* head=(struct node*)malloc(sizeof(struct node));
    struct node* second=(struct node*)malloc(sizeof(struct node));
    struct node* third=(struct node*)malloc(sizeof(struct node));

    head->data=10;
    head->next=second;
    head->prev=NULL;

    second->data=20;
    second->next=third;
    second->prev=head;

    third->data=30;
    third->next=NULL;
    third->prev=second;
    
    linklisttraversal(head);
    return 0;

}